mock-collector/build_template.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: v1
+kind: Template
+labels:
+  template: topological-inventory-mock-collector-build
+metadata:
+  name: topological-inventory-mock-collector-build
+objects:
+- apiVersion: v1
+  kind: BuildConfig
+  metadata:
+    name: topological-inventory-mock-collector
+  spec:
+    source:
+      type: Git
+      git:
+        uri: ${SOURCE_REPOSITORY_URL}
+        ref: ${SOURCE_REPOSITORY_REF}
+    resources:
+      limits:
+        memory: 1Gi
+    strategy:
+      type: Docker
+    output:
+      to:
+        kind: ImageStreamTag
+        name: topological-inventory-mock-collector:latest
+        namespace: ${IMAGE_NAMESPACE}
+    triggers:
+    - type: GitHub
+      github:
+        secret: ${GITHUB_WEBHOOK_SECRET}
+- apiVersion: v1
+  kind: ImageStream
+  metadata:
+    name: topological-inventory-mock-collector
+  spec:
+    tags:
+    - name: latest
+parameters:
+- name: IMAGE_NAMESPACE
+  displayName: Image Namespace
+  description: Namespace to push the resulting image stream tag to
+  value: buildfactory
+- name: GITHUB_WEBHOOK_SECRET
+  displayName: GitHub Webhook Secret
+  description: Github trigger secret.  A difficult to guess string encoded as part of the webhook URL.  Not encrypted.
+  generate: expression
+  from: "[a-zA-Z0-9]{40}"
+- name: SOURCE_REPOSITORY_URL
+  description: The URL of the repository with your application source code.
+  displayName: Git repository URI
+  required: true
+  value: https://github.com/slemrmartin/mock_collector
```

Comment:
@slemrmartin I'd like to transfer your repo at some point...let's discuss...I also think it needs a name change because it's specifically for openshift, and probably should be prefixed with topological-inventory or something.  @agrare Thoughts on naming?

Regardless, I'm going to merge this as is, and then we can update it when the repo is moved.

Reply:
Yeah the binary is actually named `openshift-mock-collector` so that could work for the repo name, either way they should be the same.

Reply:
@agrare @Fryguy I thought it could serve as a mock collector for more than one source type (using the same code), so I named it `mock_collector` and there will be binaries `openshift-mock-collector`, `amazon-mock-collector` .
Do you think it's worth to implement mock-collector for other source type?

If so, there are two choices: all as 1 gem or extract common classes to separate gem (classes structure is prepared for it)